src/internal/partial_solution.rs

```diff
@@ -77,13 +86,23 @@ impl<P: Package, V: Version> PartialSolution<P, V> {
         let pos = self
             .history
             .iter()
-            .rposition(|(l, _)| *l == decision_level)
+            .rposition(
+                |AssignmentDepth {
+                     decision_level: l,
+                     assignment: _,
+                 }| *l == decision_level,
+            )
             .unwrap_or(self.history.len() - 1);
         *self = Self::from_assignments(
             std::mem::take(&mut self.history)
                 .into_iter()
                 .take(pos + 1)
-                .map(|(_, a)| a),
+                .map(
+                    |AssignmentDepth {
+                         decision_level: _,
+                         assignment,
+                     }| assignment,
```

Comment:
I'm not certain, but I believe you can write something like `|AssignmentDepth { assignment, .. }| `

Reply:
same for the change above on rposition

Reply:
Or even better, we can just do `.map(|assignment_depth| assignment_depth.assignment)`. Thanks for bringing my attention to these lines.

Reply:
Changed both in the latest commit "refactor: shorten AssignmentDepth matching".